vowpalwabbit/cb_explore_adf_common.h

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright (c) by respective owners including Yahoo!, Microsoft, and
+individual contributors. All rights reserved.  Released under a BSD
+license as described in the file LICENSE.
+*/
+#pragma once
+#include <stdint.h>
+#include "v_array.h"
+
+#include "learner.h"
+#include "action_score.h"
```

Comment:
Minimize dependencies with forward declarations

Reply:
Common is templatized now, so it requires full implementation in the header.
All other reductions have minimized headers